shm/Map.cpp

```diff
@@ -0,0 +1,48 @@
+#include "Map.hpp"
+#include <algorithm>
+
+coordinateType generatePosition() {
+    std::random_device rd;
+    std::mt19937 gen(rd());
+    std::uniform_int_distribution<> distrib(0, 10);
```

Comment:
I think this should be a member of the class. Creating a random device based on `mt19937` take time, and also can make some problems, because we have a limited resource of pseudo numbers.